src/sentry/static/sentry/app/components/avatarSettings.jsx

```diff
@@ -0,0 +1,515 @@
+    let resizeDimensions = this.state.resizeDimensions;
+    let diff = ((yDiff - (yDiff * 2)) + (xDiff - (xDiff * 2))) / 2;
+    let size = resizeDimensions.size + diff;
+    let left = resizeDimensions.left - diff;
+    let top = resizeDimensions.top - diff;
+    let maxSize = Math.min($container.width() - left,
+                           $container.height() - top);
+    if (top < 0) {
+      size = size + top;
+      left = left - top;
+      top = 0;
+    }
+    if (left < 0) {
+      size = size + left;
+      top = top - left;
+      left = 0;
+    }
+    if (size > maxSize) {
+      top = top + size - maxSize;
+      left = left + size - maxSize;
+      size = maxSize;
+    } else if (size < this.MIN_DIMENSION) {
+      top = top + size - this.MIN_DIMENSION;
+      left = left + size - this.MIN_DIMENSION;
+      size = this.MIN_DIMENSION;
+    }
+    return {size: size, top: top, left: left};
+  },
+
+  getUpdatedSizeNE($container, yDiff, xDiff) {
+    let resizeDimensions = this.state.resizeDimensions;
+    let diff = ((yDiff - (yDiff * 2)) + xDiff) / 2;
+    let size = resizeDimensions.size + diff;
+    let top = resizeDimensions.top - diff;
+    let maxSize = Math.min($container.width() - resizeDimensions.left,
+                           $container.height() - top);
+    if (top < 0) {
+      size = size + top;
+      top = 0;
+    }
+    if (size > maxSize) {
+      top = top + size - maxSize;
+      size = maxSize;
+    } else if (size < this.MIN_DIMENSION) {
+      top = top + size - this.MIN_DIMENSION;
+      size = this.MIN_DIMENSION;
+    }
+    return Object.assign({}, resizeDimensions, {size: size, top: top});
+  },
+
+  getUpdatedSizeSW($container, yDiff, xDiff) {
+    let resizeDimensions = this.state.resizeDimensions;
+    let diff = (yDiff + (xDiff - (xDiff * 2))) / 2;
+    let size = resizeDimensions.size + diff;
+    let left = resizeDimensions.left - diff;
+    let maxSize = Math.min($container.width() - left,
+                           $container.height() - resizeDimensions.top);
+    if (left < 0) {
+      size = size + left;
+      left = 0;
+    }
+    if (size > maxSize) {
+      left = left + size - maxSize;
+      size = maxSize;
+    } else if (size < this.MIN_DIMENSION) {
+      left = left + size - this.MIN_DIMENSION;
+      size = this.MIN_DIMENSION;
+    }
+    return Object.assign({}, resizeDimensions, {size: size, left: left});
+  },
+
+  getUpdatedSizeSE($container, yDiff, xDiff) {
+    let diff = (yDiff + xDiff) / 2;
+    let resizeDimensions = this.state.resizeDimensions;
+    let size = resizeDimensions.size + diff;
+    let maxSize = Math.min($container.width() - resizeDimensions.left,
+                           $container.height() - resizeDimensions.top);
+    if (size > maxSize) {
+      size = maxSize;
+    } else if (size < this.MIN_DIMENSION) {
+      size = this.MIN_DIMENSION;
+    }
+    return Object.assign({}, resizeDimensions, {size: size});
+  },
```

Comment:
Hmm, these methods sure look awfully similar. Is there a way we can reduce them into a common function/method?